packages/react-checkbox/src/components/Checkbox/useCheckboxStyles.ts

```diff
@@ -1,26 +1,245 @@
+    theme => ({
+      ':after': {
+        content: "''",
+        position: 'absolute',
+        width: '100%',
+        height: '100%',
+        border: `2px solid ${theme.alias.color.neutral.neutralForeground1}`,
+        borderRadius: '4px',
+        margin: '-6px',
+      },
+    }),
+    { selector: 'focus-within' },
+  ),
+});
+
+const useInputStyle = makeStyles({
+  input: {
+    opacity: 0,
+    margin: 0,
+    padding: 0,
+    cursor: 'pointer',
+  },
+
+  disabled: {
+    cursor: 'default',
+  },
+});
+
+const useBoxStyles = makeStyles({
+  box: theme => ({
+    display: 'flex',
+    flexShrink: 0,
+    alignItems: 'center',
+    justifyContent: 'center',
+    boxSizing: 'border-box',
+    borderStyle: 'solid',
+    borderRadius: theme.global.borderRadius.small,
+  }),
+
+  medium: {
+    width: '16px',
+    height: '16px',
+  },
+
+  large: {
+    width: '20px',
+    height: '20px',
+  },
+
+  // TODO: change marginLeft to Spacing horizontal M once it's added
+  before: theme => ({
+    marginLeft: '12px',
+  }),
+
+  // TODO: change marginRight to Spacing horizontal M once it's added
+  after: theme => ({
+    marginRight: '12px',
+  }),
+
+  circular: theme => ({
+    borderRadius: theme.global.borderRadius.circular,
+  }),
+
+  disabled: theme => ({
+    borderWidth: theme.global.strokeWidth.thin,
+    borderColor: theme.alias.color.neutral.neutralStrokeDisabled,
+  }),
+
+  unchecked: theme => ({
+    borderColor: theme.alias.color.neutral.neutralStrokeAccessible,
+    borderWidth: theme.global.strokeWidth.thin,
+    ':hover': {
+      borderColor: theme.alias.color.neutral.neutralStrokeAccessibleHover,
+    },
+    ':active': {
+      borderColor: theme.alias.color.neutral.neutralStrokeAccessiblePressed,
+    },
+  }),
+
+  checked: theme => ({
+    backgroundColor: theme.alias.color.neutral.compoundBrandBackground,
+    borderWidth: 0,
+    ':hover': {
+      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundHover,
+    },
+    ':active': {
+      backgroundColor: theme.alias.color.neutral.compoundBrandBackgroundPressed,
+    },
+  }),
+
+  mixed: theme => ({
+    borderColor: theme.alias.color.neutral.compoundBrandStroke,
+    borderWidth: theme.global.strokeWidth.thin,
+    ':hover': {
+      borderColor: theme.alias.color.neutral.compoundBrandStrokeHover,
+    },
+    ':active': {
+      borderColor: theme.alias.color.neutral.compoundBrandStrokePressed,
+    },
+  }),
+});
+
+const useIndicatorStyles = makeStyles({
+  indicator: {
+    position: 'absolute',
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+
+  // TODO: Remove fontSize once checkbox uses react-icons
+  medium: {
+    fontSize: '8px',
+    width: '8px',
+    height: '8px',
+  },
+
+  // TODO: Remove fontSize once checkbox uses react-icons
+  large: {
+    fontSize: '10px',
+    width: '10px',
+    height: '10px',
+  },
+
+  // TODO: Remove fill once checkbox uses react-icons
```

Comment:
Sorry to go back and forth on this... to make this more portable to work with any SVG, you can change these `disabled`, `checked`, etc. states to set `color` only, and then add this to the root `indicator` style (line 126):
```ts
fill: 'currentColor',
```

Then you can remove the TODO comment as well.